### DISCUSSION TITLE
Is there any way to get the model name from the generated code?

### DISCUSSION THREAD

**Discussion by moznion** [prisma/prisma#21497] [Q&A]
### Question

When it has the following model in `schema.prisma`

```prisma
model User {
 ...
}
```

Then Prisma generates the following code 

```TypeScript
...
export type User = $Result.DefaultSelection<Prisma.$UserPayload>
...
export class PrismaClient<
  T extends Prisma.PrismaClientOptions = Prisma.PrismaClientOptions,
  U = 'log' extends keyof T ? T['log'] extends Array<Prisma.LogLevel | Prisma.LogDefinition> ? Prisma.GetEvents<T['log']> : never : never,
  ExtArgs extends $Extensions.Args = $Extensions.DefaultArgs
> {
  ...
  export interface UserDelegate<ExtArgs extends $Extensions.Args = $Extensions.DefaultArgs> {
    [K: symbol]: { types: Prisma.TypeMap<ExtArgs>['model']['User'], meta: { name: 'User' } }
    ...
  }
  ...
}
...
export namespace Prisma {
  ...
  export type $UserPayload<ExtArgs extends $Extensions.Args = $Extensions.DefaultArgs> = {
    name: "User"
    objects: {}
    scalars: $Extensions.GetResult<{
      ...
    }, ExtArgs["result"]["user"]>
    composites: {}
  }
  ...
}
```

It seems there are the model name `User` in several places, but I couldn't find the method to get that model name through the code as programmatically. Is it possible to retrieve that model name from the generated code?

**Comment by ludralph** [comment]
Hi @moznion 👋 

Thank you for raising this question.

Can you please elaborate on your use case for why you need to get a model by its name?

You can access all models using `Prisma.dmmf.datamodel.models` but you should know that the dmmf is an internal api which is subject to change.

The output from Prisma.dmmf.datamodel.models is of the form:
```js
[
  {
    name: 'Pet',
    dbName: null,
    fields: [ [Object], [Object], [Object], [Object], [Object] ],
    primaryKey: null,
    uniqueFields: [],
    uniqueIndexes: [],
    isGenerated: false
  },
  {
    name: 'Owner',
    dbName: null,
    fields: [ [Object], [Object], [Object], [Object] ],
    primaryKey: null,
    uniqueFields: [],
    uniqueIndexes: [],
    isGenerated: false
  }
]
```
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
